### PR TITLE
Add user store regEx validations in users/roles/groups sections

### LINKED ISSUE #875: Validate username, role name, password against user store regular expressions
**Is your feature request related to a problem? Please describe.**
Currently, in the user/group/role creation and edit flow, we don't validate the user inputs like username, password, and role name in the developer portal. 

**Describe the solution you'd like**
It's better to validate the above-mentioned input against the user store specific regExes.

### PR DESCRIPTION
**Pull request by NipuniBhagya**
Resolves #875

### CODE REVIEW

File: apps/developer-portal/src/components/roles/create-role-wizard/role-basics.tsx [hunk 1]
```diff
@@ -100,14 +110,43 @@ export const RoleBasics: FunctionComponent<RoleBasicProps> = (props: RoleBasicPr
...
+        const domain: string = values.get("domain").toString();
+        setUserStore(domain);
+    };
+
+    /**
+     * The following function validates role name against the user store regEx.
+     *
+     * @param roleName - User input role name
+     */
+    async function validateRoleNamePattern(roleName: string) {
```
Comment (brionmario) [inline apps/developer-portal/src/components/roles/create-role-wizard/role-basics.tsx:136]: Shall we use an es6 async function here? Check other places as well.

```suggestion
    const validateRoleNamePattern = async (roleName: string) => {
```

File: apps/developer-portal/src/components/roles/edit-role/edit-role-basic.tsx [hunk 2]
```diff
@@ -57,15 +67,58 @@ export const BasicRoleDetails: FunctionComponent<BasicRoleProps> = (props: Basic
...
+        return regEx;
+    }
+
+    /**
+     * The following function handles the role name change.
+     *
+     * @param event
+     * @param data
+     */
+    const handleRoleNameChange = (event: ChangeEvent, data: InputOnChangeData) => {
```
Comment (brionmario) [inline apps/developer-portal/src/components/roles/edit-role/edit-role-basic.tsx:118]:
```suggestion
    const handleRoleNameChange = (event: ChangeEvent, data: InputOnChangeData): void => {
```

File: apps/developer-portal/src/components/roles/create-role-wizard/role-basics.tsx (hunk 1 above)
Comment (NipuniBhagya) [reply apps/developer-portal/src/components/roles/create-role-wizard/role-basics.tsx]: Resolved with 153fa58

File: apps/developer-portal/src/components/roles/edit-role/edit-role-basic.tsx (hunk 2 above)
Comment (NipuniBhagya) [reply apps/developer-portal/src/components/roles/edit-role/edit-role-basic.tsx]: Resolved with 153fa58

**Pull request merged**